pypdf/_text_extraction/_layout_mode/_font.py

```diff
@@ -92,6 +92,11 @@ def __post_init__(self) -> None:
                             }
                         )
                         skip_count = 2
+                    else:
+                        raise PdfReadError(
```

Comment:
Does this need to be a hard error or could we continue with just `logger_warning` in this case?

Reply:
This is my first contribution so take my opinion with a grain of salt. As a user of this package, I would rather a malformed PDF raise an error than a quiet warning. I can always catch the error and handle it as I wish (in this case, I would fall back on non-layout mode text extraction). If it's a logger warning it's not always obvious that something went wrong, at all.

If we do simply warn, what would we do? Just proceed to the next index and try again? I think the end result there is a random font extraction that doesn't make sense.

Reply:
Thanks for the explanation. It apparently worked without this error in the past and this seems to only be about character widths(?), thus I would not see this as a hard error.

Nevertheless, I am open to further discussions and comments on this. Once this is out of the WIP/draft state, @shartzog who wrote the initial implementation might provide some further insights, clearly knowing more about this specific code than me.

Reply:
I suspect that this worked in the past because no one ever tested it with a malformed PDF, or if they did, it just silently misbehaved in an poorly defined manner.

For what it's worth, this started for me with a TypeError (Int + IndirectObject), which would be fixed just by the addition of `get_object()` above. The rest was me trying to clean up what struck me as a poorly defined error handling for when the widths were not as expected. I don't feel strongly about this: the IndirectObject fix is what's critical to me.

Reply:
The code example you've provided looks good : can you add a test using them to confirm no exception is raised?

Reply:
If we do not want to add `logger_warning` here, could we at least use some specific exception subclass instead to better differentiate whether this is hard reading error or just some "smaller" spec violation?

Reply:
Does an appropriate one exist, or should I create one?

Reply:
Also, regarding the test: can I have some guidance as to where you would want this to go? I do not feel capable of writing a test from scratch from this package so would ideally want a similar test to use as a template.

Reply:
I used ParseError and wrote a test (both positive and negative).